File: language-samples/hello.clj
; Clojure sample for GitHub language detection
; BarbrickDesign - Complete Language Portfolio

(ns hello
  (:gen-class))

(defrecord Greeter [message])

(defn greet [greeter]
  (println (:message greeter)))

(defn -main []
  (let [greeter (->Greeter "Hello from Clojure!")]
    (greet greeter)
    
    ; Vector
    (let [numbers [1 2 3 4 5]
          doubled (map #(* % 2) numbers)]
      (println doubled))
    
    ; Map
    (let [info {:language "Clojure"
                :paradigm "Functional"
                :typing "Dynamic"}]
      (println info))
    
    ; Collection functions
    (let [languages ["clojure" "scala" "erlang"]]
      (doseq [lang languages]
        (println (.toUpperCase lang))))
    
    ; Thread-first macro
    (-> [1 2 3 4 5]
        (map inc)
        (filter even?)
        println)))

(-main)

File: language-samples/hello.dart
// Dart sample for GitHub language detection
// BarbrickDesign - Complete Language Portfolio

class Greeter {
  final String message;
  
  Greeter(this.message);
  
  void greet() {
    print(message);
  }
}

void main() {
  var greeter = Greeter("Hello from Dart!");
  greeter.greet();
  
  // List
  var numbers = [1, 2, 3, 4, 5];
  var doubled = numbers.map((n) => n * 2).toList();
  
  // Map
  var info = {
    'language': 'Dart',
    'paradigm': 'Multi-paradigm',
    'typing': 'Static',
  };
  
  // Collection methods
  var languages = ['Dart', 'Flutter', 'JavaScript'];
  languages.forEach((lang) {
    print(lang.toUpperCase());
  });
  
  // Async/await
  Future<String> fetchData() async {
    await Future.delayed(Duration(seconds: 1));
    return 'Data loaded';
  }
}
